### PR TITLE
Improve getRoles service of Role v2 management

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
### Proposed changes in this pull request

Improved getRoles serviceof Role v2 management.

Related Issue
- https://github.com/wso2/product-is/issues/18096

Related PRs

- https://github.com/wso2/charon/pull/397
- https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/515

### CODE REVIEW

**Review by shashimalcse — APPROVED** [review]

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementService.java [hunk 1]
```diff
@@ -55,32 +55,33 @@ RoleBasicInfo addRole(String roleName, List<String> userList, List<String> group
...
-     * @param offset       Offset value.
-     * @param sortBy       SortBy value.
-     * @param sortOrder    Sort order value.
-     * @param tenantDomain Tenant domain.
+     * @param limit              Limit value.
+     * @param offset             Offset value.
+     * @param sortBy             SortBy value.
+     * @param sortOrder          Sort order value.
+     * @param tenantDomain       Tenant domain.
+     * @param requiredAttributes Required attributes
```
Comment (dewniMW) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementService.java:63]:
```suggestion
     * @param requiredAttributes Required attributes.
```

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementService.java [hunk 2]
```diff
@@ -55,32 +55,33 @@ RoleBasicInfo addRole(String roleName, List<String> userList, List<String> group
...
-     * @param sortBy       SortBy value.
-     * @param sortOrder    Sort order value.
-     * @param tenantDomain Tenant domain.
+     * @param filter             Filter for the Role ID.
+     * @param limit              Limit value.
+     * @param offset             Offset value.
+     * @param sortBy             SortBy value.
+     * @param sortOrder          Sort order value.
+     * @param tenantDomain       Tenant domain.
+     * @param requiredAttributes Required attributes
```
Comment (dewniMW) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementService.java:79]:
```suggestion
     * @param requiredAttributes Required attributes.
```

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementService.java (hunk 1 above)
Comment (ChanikaRuchini) [reply components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementService.java]: fixed

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementService.java (hunk 2 above)
Comment (ChanikaRuchini) [reply components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementService.java]: fixed

**Review by dewniMW — APPROVED** [review]

**Comment by sadilchamishka** [comment]
Screenshot 2023-12-11 at 07 28 38

**Pull request merged**